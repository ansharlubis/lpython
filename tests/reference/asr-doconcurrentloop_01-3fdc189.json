{
    "basename": "asr-doconcurrentloop_01-3fdc189",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/doconcurrentloop_01.py",
    "infile_hash": "cbb1e60c50c639cdbd0e94f5aa6f6c4a8532438c52d77257ac9ccd1e",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-doconcurrentloop_01-3fdc189.stdout",
    "stdout_hash": "9b5de6bed77721fbf2b6df2612b693f1facbce6c51473e4da619a85c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}