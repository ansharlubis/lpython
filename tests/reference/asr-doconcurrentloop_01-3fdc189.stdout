([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         main0: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m10000[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9999[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m10000[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9999[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     c: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           c [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m10000[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9999[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     i: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     nsize: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           nsize [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     scalar: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           scalar [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               main0 [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 3 [33mscalar[39m) 
                  ([1m[35mCast[39m[0m 
                     ([1m[35mRealConstant[39m[0m   1.00000000000000000e+01 
                        ([1m[35mReal[39m[0m 8 [])) [1m[32m
                     RealToReal[39m[0m 
                     ([1m[35mReal[39m[0m 4 []) 
                     ([1m[35mRealConstant[39m[0m   1.00000000000000000e+01 
                        ([1m[35mReal[39m[0m 4 []))) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 3 [33mnsize[39m) ([1m[35mIntegerConstant[39m[0m [36m1234[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35mDoConcurrentLoop[39m[0m 
                  (
                     ([1m[35mVar[39m[0m 3 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mVar[39m[0m 3 [33mnsize[39m) [1m[32m
                        Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))) [
                  ([1m[35m=[39m[0m 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 3 [33ma[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 3 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 4 []) ()) 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) 
                        ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 4 []))) ()) 
                  ([1m[35m=[39m[0m 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 3 [33mb[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 3 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 4 []) ()) 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) 
                        ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 4 []))) ())]) 
               ([1m[35mSubroutineCall[39m[0m 1 [33mtriad[39m () [
                  (
                     ([1m[35mVar[39m[0m 3 [33ma[39m)) 
                  (
                     ([1m[35mVar[39m[0m 3 [33mb[39m)) 
                  (
                     ([1m[35mVar[39m[0m 3 [33mscalar[39m)) 
                  (
                     ([1m[35mVar[39m[0m 3 [33mc[39m))] ()) 
               ([1m[35mPrint[39m[0m () [
                  ([1m[35mStringConstant[39m[0m "End Stream Triad" 
                     ([1m[35mCharacter[39m[0m 1 16 () []))] () ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.), 
         triad: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     N: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           N [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           InOut[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (() ())]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           b [1m[32m
                           InOut[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (() ())]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     c: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           c [1m[32m
                           InOut[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 [
                              (() ())]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     i: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     scalar: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           scalar [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               triad [
               ([1m[35mVar[39m[0m 2 [33ma[39m) 
               ([1m[35mVar[39m[0m 2 [33mb[39m) 
               ([1m[35mVar[39m[0m 2 [33mscalar[39m) 
               ([1m[35mVar[39m[0m 2 [33mc[39m)] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mN[39m) ([1m[35mIntegerConstant[39m[0m [36m1234[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35mDoConcurrentLoop[39m[0m 
                  (
                     ([1m[35mVar[39m[0m 2 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mN[39m) [1m[32m
                        Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))) [
                  ([1m[35m=[39m[0m 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mc[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 4 []) ()) 
                     ([1m[35mRealBinOp[39m[0m 
                        ([1m[35mArrayItem[39m[0m 
                           ([1m[35mVar[39m[0m 2 [33ma[39m) [
                           (() 
                              ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                           ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                        Add[39m[0m 
                        ([1m[35mRealBinOp[39m[0m 
                           ([1m[35mVar[39m[0m 2 [33mscalar[39m) [1m[32m
                           Mul[39m[0m 
                           ([1m[35mArrayItem[39m[0m 
                              ([1m[35mVar[39m[0m 2 [33mb[39m) [
                              (() 
                                 ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                              ([1m[35mReal[39m[0m 4 []) ()) 
                           ([1m[35mReal[39m[0m 4 []) ()) 
                        ([1m[35mReal[39m[0m 4 []) ()) ())])] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mmain0[39m () [] ())])
(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 5 {}) _lpython_main_program [] [(SubroutineCall 1 main0 () [] ())] Source Public Implementation () .false. .false.), main0: (Subroutine (SymbolTable 3 {a: (Variable 3 a Local () () Default (Real 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 10000 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9999 (Integer 4 []))))]) Source Public Required .false.), b: (Variable 3 b Local () () Default (Real 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 10000 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9999 (Integer 4 []))))]) Source Public Required .false.), c: (Variable 3 c Local () () Default (Real 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 10000 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9999 (Integer 4 []))))]) Source Public Required .false.), i: (Variable 3 i Local () () Default (Integer 4 []) Source Public Required .false.), nsize: (Variable 3 nsize Local () () Default (Integer 4 []) Source Public Required .false.), scalar: (Variable 3 scalar Local () () Default (Real 4 []) Source Public Required .false.)}) main0 [] [(= (Var 3 scalar) (Cast (RealConstant   1.00000000000000000e+01 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   1.00000000000000000e+01 (Real 4 []))) ()) (= (Var 3 nsize) (IntegerConstant 1234 (Integer 4 [])) ()) (DoConcurrentLoop ((Var 3 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (Var 3 nsize) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (ArrayItem (Var 3 a) [(() (Var 3 i) ())] (Real 4 []) ()) (Cast (RealConstant   5.00000000000000000e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   5.00000000000000000e+00 (Real 4 []))) ()) (= (ArrayItem (Var 3 b) [(() (Var 3 i) ())] (Real 4 []) ()) (Cast (RealConstant   5.00000000000000000e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   5.00000000000000000e+00 (Real 4 []))) ())]) (SubroutineCall 1 triad () [((Var 3 a)) ((Var 3 b)) ((Var 3 scalar)) ((Var 3 c))] ()) (Print () [(StringConstant "End Stream Triad" (Character 1 16 () []))] () ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 4 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), triad: (Subroutine (SymbolTable 2 {N: (Variable 2 N Local () () Default (Integer 4 []) Source Public Required .false.), a: (Variable 2 a InOut () () Default (Real 4 [(() ())]) Source Public Required .false.), b: (Variable 2 b InOut () () Default (Real 4 [(() ())]) Source Public Required .false.), c: (Variable 2 c InOut () () Default (Real 4 [(() ())]) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), scalar: (Variable 2 scalar In () () Default (Real 4 []) Source Public Required .false.)}) triad [(Var 2 a) (Var 2 b) (Var 2 scalar) (Var 2 c)] [(= (Var 2 N) (IntegerConstant 1234 (Integer 4 [])) ()) (DoConcurrentLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (Var 2 N) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (ArrayItem (Var 2 c) [(() (Var 2 i) ())] (Real 4 []) ()) (RealBinOp (ArrayItem (Var 2 a) [(() (Var 2 i) ())] (Real 4 []) ()) Add (RealBinOp (Var 2 scalar) Mul (ArrayItem (Var 2 b) [(() (Var 2 i) ())] (Real 4 []) ()) (Real 4 []) ()) (Real 4 []) ()) ())])] Source Public Implementation () .false. .false.)}) [])
